internal/services/redhatopenshift/redhat_openshift_cluster_resource.go

```diff
@@ -0,0 +1,782 @@
+	"github.com/hashicorp/go-azure-helpers/lang/response"
+	"github.com/hashicorp/go-azure-helpers/resourcemanager/commonids"
+	"github.com/hashicorp/go-azure-helpers/resourcemanager/commonschema"
+	"github.com/hashicorp/go-azure-helpers/resourcemanager/location"
+	"github.com/hashicorp/go-azure-sdk/resource-manager/redhatopenshift/2023-09-04/openshiftclusters"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-provider-azurerm/helpers/azure"
+	"github.com/hashicorp/terraform-provider-azurerm/helpers/tf"
+	commonValidate "github.com/hashicorp/terraform-provider-azurerm/helpers/validate"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/sdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/services/redhatopenshift/validate"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/suppress"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/validation"
+)
+
+var _ sdk.ResourceWithUpdate = RedHatOpenShiftCluster{}
+
+type RedHatOpenShiftCluster struct{}
+
+type RedHatOpenShiftClusterModel struct {
+	Name             string             `tfschema:"name"`
+	Location         string             `tfschema:"location"`
+	ResourceGroup    string             `tfschema:"resource_group_name"`
+	ConsoleUrl       string             `tfschema:"console_url"`
+	ServicePrincipal []ServicePrincipal `tfschema:"service_principal"`
+	ClusterProfile   []ClusterProfile   `tfschema:"cluster_profile"`
+	NetworkProfile   []NetworkProfile   `tfschema:"network_profile"`
+	MainProfile      []MainProfile      `tfschema:"main_profile"`
+	WorkerProfile    []WorkerProfile    `tfschema:"worker_profile"`
+	ApiServerProfile []ApiServerProfile `tfschema:"api_server_profile"`
+	IngressProfile   []IngressProfile   `tfschema:"ingress_profile"`
+	Tags             map[string]string  `tfschema:"tags"`
+}
+
+type ServicePrincipal struct {
+	ClientId     string `tfschema:"client_id"`
+	ClientSecret string `tfschema:"client_secret"`
+}
+
+type ClusterProfile struct {
+	PullSecret      string `tfschema:"pull_secret"`
+	Domain          string `tfschema:"domain"`
+	FipsEnabled     bool   `tfschema:"fips_enabled"`
+	ResourceGroupId string `tfschema:"resource_group_id"`
+	Version         string `tfschema:"version"`
+}
+
+type NetworkProfile struct {
+	OutboundType string `tfschema:"outbound_type"`
+	PodCidr      string `tfschema:"pod_cidr"`
+	ServiceCidr  string `tfschema:"service_cidr"`
+}
+
+type MainProfile struct {
+	SubnetId                string `tfschema:"subnet_id"`
+	VmSize                  string `tfschema:"vm_size"`
+	EncryptionAtHostEnabled bool   `tfschema:"encryption_at_host_enabled"`
+	DiskEncryptionSetId     string `tfschema:"disk_encryption_set_id"`
+}
+
+type WorkerProfile struct {
+	VmSize                  string `tfschema:"vm_size"`
+	DiskSizeGb              int64  `tfschema:"disk_size_gb"`
```

Comment:
(just passing through) but FWIW these should be `int64` and not `int` - since we're expecting int64's everywhere else (SDK etc) even if that's downcast to a smaller variable size

Reply:
Hi @tombuildsstuff, in a previous PR, @jackofallops told us to use `int` for such property as other existing `int64-type` properties in the provider. And according to [this code snippet](https://github.com/hashicorp/terraform-provider-azurerm/blob/fc0bfa18efa74a94c24b12eda48b00cb508c7d22/vendor/github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema/field_reader.go#L325), `TypeInt` is parsed as `int`. Could you kindly help double confirm whether we need to use `int64` for the `int64-type` properties of model in future?